rolocController.cpp

```diff
@@ -197,50 +224,62 @@ void ROLOCcontroller::processRolocData()
-            qDebug() << "Standard Dev.: " << stdDev;
+            // update the sig strength
+            mROLOCsignalStrenth = rolocData;
+#if DBG_BLOCK
+            qDebug() << "sig strength = " << mROLOCsignalStrenth; // TODO verify if is this working ?
+#endif
+            // clear the depth measurements
+            mDepthAccumulator.clear();
+            sendDataReport();
+        }
+        else
+        if (mCurrentMode == ROLOC::eMODE_GET_DEPTH_MEASUREMENT)
+        {
+            mDepthAccumulator.append(rolocData);
+            mNumSamples++;
 
-            for(quint8 i = 0; i < mDepthAccumulator.count(); i++)
+            if(mNumSamples > N_MULTI_LF_DEPTH_SAMPLE)
             {
-                quint8 isLessThanLowerBound = mDepthAccumulator.at(i) < mean - stdDev * scaleOfElimination;
-                quint8 isGreaterThanUpperBound = mDepthAccumulator.at(i) > mean + stdDev * scaleOfElimination;
-                quint8 isOutOfBounds = isLessThanLowerBound || isGreaterThanUpperBound;
+                QList<quint8> acceptedDepthReadings;
+                float scaleOfElimination = (float)N_MULTI_LF_DEPTH_DELTA;
 
-                if(!isOutOfBounds)
+                double mean = getMean(mDepthAccumulator);
+                double stdDev = qSqrt(getVariance(mDepthAccumulator));
+#if DBG_BLOCK
+                qDebug() << "*************************Distance Stats*****************";
+                qDebug() << "*************************** bit set  -> " <<  CHECK_BIT(rolocData, 1)  << "*********************";
+                qDebug() << "smaple size: " << mNumSamples;
+                qDebug() << "sample size: " << mNumSamples;
+                qDebug() << "Mean: " << mean;
+                qDebug() << "Standard Dev.: " << stdDev;
+#endif
+                for(quint8 i = 0; i < mDepthAccumulator.count(); i++)
                 {
-                    acceptedDepthReadings.append(mDepthAccumulator.at(i));
+                    quint8 isLessThanLowerBound = mDepthAccumulator.at(i) < mean - stdDev * scaleOfElimination;
+                    quint8 isGreaterThanUpperBound = mDepthAccumulator.at(i) > mean + stdDev * scaleOfElimination;
+                    quint8 isOutOfBounds = isLessThanLowerBound || isGreaterThanUpperBound;
+
+                    if(!isOutOfBounds)
+                    {
+                        acceptedDepthReadings.append(mDepthAccumulator.at(i));
+                    }
                 }
-            }
 
-            double finalReading = getMean(acceptedDepthReadings);
-            qDebug() << finalReading;
-            mROLOCdepthMeasurement = finalReading;
+                double finalReading = getMean(acceptedDepthReadings);
+                mROLOCdepthMeasurement = finalReading  * 0.0254;
+                sendDataReport();
+                mNumSamples = 0;
+                mCurrentMode = ROLOC::eMODE_GET_SIGNAL_STRENGTH;
+                qDebug() << "mFrequency  mFrequency-> " << mFrequency;
```

Comment:
lol this debug looks redundant
```suggestion
                qDebug() << "mFrequency-> " << mFrequency;
```